src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/GovNotifyEmailService.kt

```diff
@@ -1,11 +1,59 @@
 package uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service
 
+import org.slf4j.LoggerFactory
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.config.Email
 import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.config.EmailService
 import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.config.EmailTemplates
 import uk.gov.service.notify.NotificationClient
 
-internal class GovNotifyEmailService(client: NotificationClient, emailTemplates: EmailTemplates) : EmailService {
-  override fun sendEmail() {
-    TODO("Service is not yet implemented")
+class GovNotifyEmailService(
+  private val client: NotificationClient,
+  private val emailTemplates: EmailTemplates,
+) : EmailService {
+
+  companion object {
+    private val log = LoggerFactory.getLogger(this::class.java)
+  }
+
+  override fun send(email: Email) {
+    when (email) {
+      is AppointmentCancelledCourtEmail -> send(email, emailTemplates.appointmentCancelledCourt)
```

Comment:
We could introduce an enum type that is shared between email impl and the email templates to allow us to pull out the template by type.  This `when` technically would become redundant e.g.

`send(email, emailTemplates.getTemplate(email.templateType)`

Reply:
I think the `when` is fine - clear what is happening.